apps/dashboard/app/(app)/settings/root-keys/new/client.tsx

```diff
@@ -113,7 +113,7 @@ export const Client: React.FC<Props> = ({ apis }) => {
     // for large data sets.
 
     setCardStatesMap(initialCardStates);
-  }, []); // Execute ones on the first load
+  }, [apis, selectedPermissions]); // Execute ones on the first load
```

Comment:
the comment explicitly states that the effect should only run once. Adding dependencies breaks this intended behavior.